core/src/desul/atomics/SYCLConversions.hpp

```diff
@@ -9,50 +9,92 @@ SPDX-License-Identifier: (BSD-3-Clause)
+  static constexpr sycl_memory_order value = sycl_memory_order::seq_cst;
 };
-template<>
+template <>
 struct DesulToSYCLMemoryOrder<MemoryOrderAcquire> {
-  static constexpr DESUL_SYCL_NAMESPACE::memory_order value = DESUL_SYCL_NAMESPACE::memory_order::acquire;
+  static constexpr sycl_memory_order value = sycl_memory_order::acquire;
 };
-template<>
+template <>
 struct DesulToSYCLMemoryOrder<MemoryOrderRelease> {
-  static constexpr DESUL_SYCL_NAMESPACE::memory_order value = DESUL_SYCL_NAMESPACE::memory_order::release;
+  static constexpr sycl_memory_order value = sycl_memory_order::release;
 };
-template<>
+template <>
 struct DesulToSYCLMemoryOrder<MemoryOrderAcqRel> {
-  static constexpr DESUL_SYCL_NAMESPACE::memory_order value = DESUL_SYCL_NAMESPACE::memory_order::acq_rel;
+  static constexpr sycl_memory_order value = sycl_memory_order::acq_rel;
 };
-template<>
+template <>
 struct DesulToSYCLMemoryOrder<MemoryOrderRelaxed> {
-  static constexpr DESUL_SYCL_NAMESPACE::memory_order value = DESUL_SYCL_NAMESPACE::memory_order::relaxed;
+  static constexpr sycl_memory_order value = sycl_memory_order::relaxed;
 };
 
-template<class MemoryScope>
+template <class MemoryScope>
 struct DesulToSYCLMemoryScope;
-template<>
+template <>
 struct DesulToSYCLMemoryScope<MemoryScopeCore> {
-  static constexpr DESUL_SYCL_NAMESPACE::memory_scope value = DESUL_SYCL_NAMESPACE::memory_scope::work_group;
+  static constexpr sycl_memory_scope value = sycl_memory_scope::work_group;
 };
-template<>
+template <>
 struct DesulToSYCLMemoryScope<MemoryScopeDevice> {
-  static constexpr DESUL_SYCL_NAMESPACE::memory_scope value = DESUL_SYCL_NAMESPACE::memory_scope::device;
+  static constexpr sycl_memory_scope value = sycl_memory_scope::device;
 };
-template<>
+template <>
 struct DesulToSYCLMemoryScope<MemoryScopeSystem> {
-  static constexpr DESUL_SYCL_NAMESPACE::memory_scope value = DESUL_SYCL_NAMESPACE::memory_scope::system;
+  static constexpr sycl_memory_scope value = sycl_memory_scope::system;
 };
 
-}
+// NOTE valid atomic address spaces are
+//   * access::address_space::global_space
+//   * access::address_space::local_space
+//   * access::address_space::global_device_space
+// below is an attempt to map memory scope to an address space
+template <class MemoryScope>
+struct DesulMemoryScopeToSYCLAdressSpace;
+template <>
+struct DesulMemoryScopeToSYCLAdressSpace<MemoryScopeCore> {
+  static constexpr sycl::access::address_space value =
+      sycl::access::address_space::local_space;
+};
+template <>
+struct DesulMemoryScopeToSYCLAdressSpace<MemoryScopeDevice> {
+  static constexpr sycl::access::address_space value =
+      sycl::access::address_space::global_device_space;
+};
+template <>
+struct DesulMemoryScopeToSYCLAdressSpace<MemoryScopeSystem> {
+  static constexpr sycl::access::address_space value =
+      sycl::access::address_space::global_space;
```

Comment:
```suggestion
template <>
struct DesulMemoryScopeToSYCLAdressSpace<MemoryScopeNode> {
  static constexpr sycl::access::address_space value =
      sycl::access::address_space::global_space;
template <>
struct DesulMemoryScopeToSYCLAdressSpace<MemoryScopeSystem> {
  static constexpr sycl::access::address_space value =
      sycl::access::address_space::global_space;
```
I'm not quite sure if this is correct but it seems the most reasonable.

Reply:
There is no point since `DesulToSYCLMemoryScope<MemoryScopeNode>` is not defined

Reply:
Yes, it might be sensible to add it (but that could be done later). Note that the SYCL2020 specifications define
- sycl::memory_scope::work_item
- sycl::memory_scope::sub_group
- sycl::memory_scope::work_group
- sycl::memory_scope::device
- sycl::memory_scope::system
and also says
> The broadest scope that can be applied to an atomic operation corresponds to the set of work-items which can access the associated memory location. For example, the broadest scope that can be applied to atomic operations in work-group local memory is sycl::memory_scope::work_group. If a broader scope is supplied, the behavior is as-if the narrowest scope containing all work-items which can access the associated memory location was supplied. Hence,

Hence, it doesn't harm to just specify a broad scope like `sycl::memory_scope::system` for `node`. Of course, we don't really test anything except for `device` really...